map_model/src/make/turns.rs

```diff
@@ -294,3 +297,67 @@ fn lc_penalty(t: &Turn, map: &Map) -> isize {
 
     ((from_idx as isize) - (to_idx as isize)).abs()
 }
+
+fn remove_merging_turns(map: &Map, input: Vec<Turn>, turn_type: TurnType) -> Vec<Turn> {
+    let mut turns = Vec::new();
+
+    // Group turns of the specified type by (from, to) road
+    let mut pairs: BTreeMap<(RoadID, RoadID), Vec<Turn>> = BTreeMap::new();
+    for t in input {
+        // Only do this for driving lanes
+        if !map.get_l(t.id.src).is_driving() || !map.get_l(t.id.dst).is_driving() {
+            turns.push(t);
+            continue;
+        }
+
+        if t.turn_type == turn_type {
+            pairs
+                .entry((map.get_l(t.id.src).parent, map.get_l(t.id.dst).parent))
+                .or_insert_with(Vec::new)
+                .push(t);
+        } else {
+            turns.push(t);
+        }
+    }
+
+    for (_, group) in pairs {
+        if group.len() == 1 {
+            turns.extend(group);
+            continue;
+        }
+
+        // From one to many is fine
+        if group.iter().map(|t| t.id.src).collect::<HashSet<_>>().len() == 1 {
+            turns.extend(group);
+            continue;
+        }
+
+        // TODO Handle >1 left/right turn lane.
```

Comment:
Another case to work on after unit tests (and finding a good example in OSM to validate).

Reply:
possible test case:
https://www.openstreetmap.org/way/130308599

Southbound on stewart, turning left onto 2nd:
leftmost lane is turn only, second-from-left is straight *or* left.

roadway tagged like: `turn:lanes=left|left;through|none`

I'm not sure what's conventional though.

Reply:
This one should be perfect, thanks!